### ISSUE TITLE
Mongodb replicaset fails to complete initialization 

### ISSUE THREAD

**Issue by Ariemeth** [helm/charts#2504]
**Is this a request for help?**:

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
BUG REPORT

**Version of Helm and Kubernetes**:
Helm: v2.6.2
Kubernetes: v1.8.0

**Which chart**:
mongodb-replicaset

**What happened**:
When installing the chart the pod gets stuck at Init:1/2

**What you expected to happen**:
Pods will initialize and be usable

**How to reproduce it** (as minimally and precisely as possible):
Using a fresh copy of ubuntu 16.04 fully patched up to 17Oct2017, docker 17.06.2-ce and rancher v1.6.10 install the chart with
```
helm install stable/mongodb-replicaset --set persistanceVolume.enabled=false --name=test-mongo --namespace=test
```

**Anything else we need to know**:
Running the chart up to 11Oct2017 at version 2.0.0 works fine.  This issue only seems to reproduce with the latest 2.1 chart version.

**Comment by mcedica** [comment]
I am running into the same issue. 
The problem is that the generated configMap "mongodb-replicaset"  contains a 'null' value so mongodb does not start.

Logs in the container:
`Starting a MongoDB instance...
[2017-10-17T18:47:33,217323524+0000] [on-start.sh] Waiting for MongoDB to be ready...
Unrecognized option: net.ssl`

In my case, in order to reproduce:

> helm install --name mongodb --set persistentVolume.storageClass=default  stable/mongodb-replicaset 

generates the following 'mongodb-replicaset' ConfigMap:

> apiVersion: v1
data:
  mongod.conf: |
    net:
      port: 27017
      ssl: null
    replication:
      replSetName: rs0
    security: null
    storage:
      dbPath: /data/db
kind: ConfigMap

If I modify the ConfigMap to remove the 'ssl: null' line and manually delete the init container, everything works.

**Comment by jwaldrip** [comment]
Also happening to me.

**Comment by unguiculus** [comment]
Should be fixed with #2516.
